### ISSUE TITLE
rtdetrv2推理速度

### ISSUE THREAD

**Issue by zhang123-lf** [lyuwenyu/RT-DETR#583]
博主你好，我非常喜欢你的rtdetr,我准备对他做部署，所以我非常关心他的推理速度，我使用rtdetrv2推理一张图片在cpu上耗时100ms，在cuda上耗时3000多ms.这太奇怪了，我想这远远达不到实时检测的效果。"""Copyright(c) 2023 lyuwenyu. All Rights Reserved.
"""
import sys
sys.path.append("/home/cyy/cyy/abc/RT-DETR-main/rtdetrv2_pytorch")

import torch
import torch.nn as nn 
import torchvision.transforms as T

import numpy as np 
from PIL import Image, ImageDraw

from src.core import YAMLConfig

def draw(images, labels, boxes, scores, thrh=0.6):
    for i, im in enumerate(images):
        draw = ImageDraw.Draw(im)

        scr = scores[i]
        lab = labels[i][scr > thrh]
        box = boxes[i][scr > thrh]
        scrs = scores[i][scr > thrh]

        for j, b in enumerate(box):
            draw.rectangle(list(b), outline='red')
            draw.text((b[0], b[1]), 
                     text=f"{lab[j].item()} {round(scrs[j].item(), 2)}", 
                     fill='blue')

        im.save(f'results_{i}.jpg')

def main(args):
    """main"""
    # ========== 模型加载部分 ==========
    cfg = YAMLConfig(args.config, resume=args.resume)

    if args.resume:
        checkpoint = torch.load(args.resume, map_location='cpu') 
        state = checkpoint.get('ema', checkpoint).get('module', checkpoint['model'])
    else:
        raise AttributeError('Only support resume to load model.state_dict by now.')

    cfg.model.load_state_dict(state)

    class Model(nn.Module):
        def __init__(self):
            super().__init__()
            self.model = cfg.model.deploy()
            self.postprocessor = cfg.postprocessor.deploy()
            
        def forward(self, images, orig_target_sizes):
            outputs = self.model(images)
            return self.postprocessor(outputs, orig_target_sizes)

    device = torch.device(args.device)
    model = Model().to(device).eval()  # 添加eval模式

    # ========== 数据准备部分 ==========
    im_pil = Image.open(args.im_file).convert('RGB')
    w, h = im_pil.size
    orig_size = torch.tensor([[w, h]]).to(device)  # 修正维度

    transforms = T.Compose([
        T.Resize((640, 640)),
        T.ToTensor(),
    ])
    im_data = transforms(im_pil).unsqueeze(0).to(device)

    # ========== 纯推理时间测量 ==========
    # CUDA同步（如果使用GPU）
    if device.type == 'cuda':
        torch.cuda.synchronize()
    
    start_time = time.perf_counter()  # 使用更高精度计时器
    
    with torch.no_grad():  # 禁用梯度计算
        output = model(im_data, orig_size)
    
    # 再次同步确保准确计时
    if device.type == 'cuda':
        torch.cuda.synchronize()
    
    inference_time = (time.perf_counter() - start_time) * 1000  # 转换为毫秒

    print(f"Inference time: {inference_time:.2f} ms")

    # ========== 结果处理 ==========
    labels, boxes, scores = output
    draw([im_pil], labels, boxes, scores)

if __name__ == '__main__':
    import argparse
    import time  # 添加时间模块
    
    parser = argparse.ArgumentParser()
    parser.add_argument('-c', '--config', type=str, required=True)
    parser.add_argument('-r', '--resume', type=str, required=True)
    parser.add_argument('-f', '--im-file', type=str, required=True)
    parser.add_argument('-d', '--device', type=str, 
                       default='cuda' if torch.cuda.is_available() else 'cpu')
    args = parser.parse_args()
    main(args)这是我的推理程序，在您原有的基础上计算了一下耗时.
       下面是实验结果

![Image](https://github.com/user-attachments/assets/739678ca-8f1d-4572-b8b0-0b56833a0931)

**Comment by zhang123-lf** [comment]
@lyuwenyu

**Comment by zhang123-lf** [comment]
大佬求解 @lyuwenyu

**Comment by xsa12345** [comment]
> [@lyuwenyu](https://github.com/lyuwenyu)

请问你的推理时间降下去了吗？
